hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/tagging/S3DeleteObjectTaggingRequest.java

```diff
@@ -0,0 +1,187 @@
+    OMMetrics omMetrics = ozoneManager.getMetrics();
+    omMetrics.incNumDeleteObjectTagging();
+
+    Map<String, String> auditMap = buildKeyArgsAuditMap(keyArgs);
+
+    OMResponse.Builder omResponse = OmResponseUtil.getOMResponseBuilder(
+        getOmRequest());
+
+    OMMetadataManager omMetadataManager = ozoneManager.getMetadataManager();
+    boolean acquiredLock = false;
+    OMClientResponse omClientResponse = null;
+    IOException exception = null;
+    Result result = null;
+    try {
+      mergeOmLockDetails(
+          omMetadataManager.getLock()
+              .acquireWriteLock(BUCKET_LOCK, volumeName, bucketName)
+      );
+      acquiredLock = getOmLockDetails().isLockAcquired();
+
+      validateBucketAndVolume(omMetadataManager, volumeName, bucketName);
+
+      String dbOzoneKey =
+          omMetadataManager.getOzoneKey(volumeName, bucketName, keyName);
+
+      OmKeyInfo omKeyInfo =
+          omMetadataManager.getKeyTable(getBucketLayout()).get(dbOzoneKey);
+      if (omKeyInfo == null) {
+        throw new OMException("Key not found", KEY_NOT_FOUND);
+      }
+
+      // Clear / delete the tags
+      omKeyInfo.getTags().clear();
+      // Set the UpdateID to the current transactionLogIndex
+      omKeyInfo.setUpdateID(trxnLogIndex, ozoneManager.isRatisEnabled());
+
+      // Note: Key modification time is not changed because S3 last modified
+      // time only changes when there are changes in the object content
+
+      // Update table cache
+      omMetadataManager.getKeyTable(getBucketLayout()).addCacheEntry(
+          new CacheKey<>(dbOzoneKey),
+          CacheValue.get(trxnLogIndex, omKeyInfo)
+      );
+
+      omClientResponse = new S3DeleteObjectTaggingResponse(
+          omResponse.setDeleteObjectTaggingResponse(DeleteObjectTaggingResponse.newBuilder()).build(),
+          omKeyInfo
+      );
+
+      result = Result.SUCCESS;
+    } catch (IOException ex) {
+      result = Result.FAILURE;
+      exception = ex;
+      omClientResponse = new S3DeleteObjectTaggingResponse(
+          createErrorOMResponse(omResponse, exception),
+          getBucketLayout()
+      );
+    } finally {
+      if (acquiredLock) {
+        mergeOmLockDetails(omMetadataManager.getLock()
+            .releaseWriteLock(BUCKET_LOCK, volumeName, bucketName));
+      }
+      if (omClientResponse != null) {
+        omClientResponse.setOmLockDetails(getOmLockDetails());
+      }
+    }
+
+    markForAudit(ozoneManager.getAuditLogger(), buildAuditMessage(
+        OMAction.DELETE_OBJECT_TAGGING, auditMap, exception, getOmRequest().getUserInfo()
+    ));
+
+    switch (result) {
+    case SUCCESS:
+      LOG.debug("Delete object tagging success. Volume:{}, Bucket:{}, Key:{}.", volumeName,
+          bucketName, keyName);
+      break;
+    case FAILURE:
+      omMetrics.incNumDeleteObjectTaggingFails();
+      LOG.error("Delete object tagging failed. Volume:{}, Bucket:{}, Key:{}.", volumeName,
+          bucketName, keyName, exception);
```

Comment:
I don't think we should log all failures at `ERROR` level.  System errors are OK, but requests can also fail due to input from user:

```
2024-11-05 11:27:58,742 [om1-OMStateMachineApplyTransactionThread - 0] ERROR tagging.S3DeleteObjectTaggingRequest: Delete object tagging failed. Volume:s3v, Bucket:bucket-azwyiqbquu, Key:ozone-test-2023699972/putobject/key=value/f2.
KEY_NOT_FOUND org.apache.hadoop.ozone.om.exceptions.OMException: Key not found
	at org.apache.hadoop.ozone.om.request.s3.tagging.S3DeleteObjectTaggingRequest.validateAndUpdateCache(S3DeleteObjectTaggingRequest.java:126)
	at org.apache.hadoop.ozone.protocolPB.OzoneManagerRequestHandler.lambda$0(OzoneManagerRequestHandler.java:421)
	at org.apache.hadoop.ozone.util.MetricUtil.captureLatencyNs(MetricUtil.java:46)
	at org.apache.hadoop.ozone.protocolPB.OzoneManagerRequestHandler.handleWriteRequestImpl(OzoneManagerRequestHandler.java:419)
	at org.apache.hadoop.ozone.protocolPB.RequestHandler.handleWriteRequest(RequestHandler.java:63)
	at org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine.runCommand(OzoneManagerStateMachine.java:551)
	at org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine.lambda$1(OzoneManagerStateMachine.java:367)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

Reply:
I think most OM requests will do error log for key not found.

Reply:
We should revisit that.

So far we have tried to get rid of such logging in S3 Gateway, but I think the same principle applies to OM, too.

HDDS-6206
HDDS-6247
HDDS-6559
HDDS-8011

Reply:
@adoroszlai Sure, I have added the check to prevent `KEY_NOT_FOUND` to emit error logs. We can file a new ticket to update the other OM requests or add other exception types that should not be logged.